### PR TITLE
add pytest as testing framework

### PR DESCRIPTION
**Pull request by codefritzel**
This commit adds some preparations for splitting the Python tests (see #1751). For the time being, all existing tests have been moved to py/test/legacy. In future, these tests are to be migrated to the new test structure.

All new tests should be implemented in this structure.

Depends on ganeti/ci-images#8

### CODE REVIEW

**Comment by saschalucas** [comment]
> Depends on [ganeti/ci-images#8](https://github.com/ganeti/ci-images/pull/8)

Added to

**Comment by saschalucas** [comment]
Merge of this should be delayed in favor of keeping Debian testing in CI, even if the current image is old and the new build of it broken. Maybe if we get Ubuntu 24.04 working, we can substitute Debian testing with it for the moment.

**Comment by saschalucas** [comment]
`pytest` must be `pytest-3`. Also bookworm supports `pytest-3` as a symlink.

File: test/py/legacy/pycurl_reset_unittest.py
Comment (saschalucas) [inline test/py/legacy/pycurl_reset_unittest.py:1]: This has been removed in #1783

File: test/py/legacy/cmdlib/instance_query_unittest.py
Comment (saschalucas) [inline test/py/legacy/cmdlib/instance_query_unittest.py:1]: This has been removed in #1783

File: Makefile.am [hunk 1]
```diff
@@ -1903,144 +1909,146 @@ TEST_FILES = \
...
+	test/py/legacy/ganeti.utils.storage_unittest.py \
+	test/py/legacy/ganeti.utils.text_unittest.py \
+	test/py/legacy/ganeti.utils.version_unittest.py \
+	test/py/legacy/ganeti.utils.wrapper_unittest.py \
+	test/py/legacy/ganeti.utils.x509_unittest.py \
+	test/py/legacy/ganeti.utils.bitarrays_unittest.py \
+	test/py/legacy/ganeti.utils_unittest.py \
+	test/py/legacy/ganeti.vcluster_unittest.py \
+	test/py/legacy/ganeti.workerpool_unittest.py \
+	test/py/legacy/pycurl_reset_unittest.py \
```
Comment (saschalucas) [inline Makefile.am:2014]: This has been removed in #1783

File: Makefile.am [hunk 2]
```diff
@@ -1903,144 +1909,146 @@ TEST_FILES = \
...
-	test/py/ganeti.workerpool_unittest.py \
-	test/py/qa.qa_config_unittest.py \
-	test/py/tempfile_fork_unittest.py
+	test/py/legacy/cmdlib/backup_unittest.py \
+	test/py/legacy/cmdlib/cluster_unittest.py \
+	test/py/legacy/cmdlib/cmdlib_unittest.py \
+	test/py/legacy/cmdlib/group_unittest.py \
+	test/py/legacy/cmdlib/instance_unittest.py \
+	test/py/legacy/cmdlib/instance_migration_unittest.py \
+	test/py/legacy/cmdlib/instance_query_unittest.py \
```
Comment (saschalucas) [inline Makefile.am:1925]: This has been removed in #1783

**Review by cilq — COMMENTED** [review]
This looks like a sensible approach to increase the testability of the codebase and to establish an environment that is more in line with common Python standards.

**Review by cilq — APPROVED** [review]

**Pull request merged**